### PR TITLE
logstash: Use jre11 instead of unsupported jre15 default

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Fix logstash and `nixosTests.elk.*`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/logstash/6.x.nix [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
 , makeWrapper
-, jre
+, jdk11_headless
 }:
 
 with lib;
+let
+  # https://www.elastic.co/support/matrix#logstash-and-jvm
+  jre = jdk11_headless;
+in
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/logstash/6.x.nix:13]: Do you know by chance when this approach and overring the input of jre is preferred?

Comment (roberth) [reply pkgs/tools/misc/logstash/6.x.nix:13]: idk, how else would you do it?

Comment (SuperSandro2000) [reply pkgs/tools/misc/logstash/6.x.nix:13]: I typed a bit fast and did some typos. The alternative would be to assign jre11_headless to jre in top-level.nix for logstash.

Comment (roberth) [reply pkgs/tools/misc/logstash/6.x.nix:13]: Right. I guess that would be easier to override.

**Comment by roberth** [comment]
This fix is ready for review.

File: pkgs/tools/misc/logstash/7.x.nix [hunk 2]
```diff
@@ -3,12 +3,13 @@
...
 , fetchurl
 , makeWrapper
+, nixosTests
 , jre
 }:
 
 with lib;
 
-stdenv.mkDerivation rec {
+let this = stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/logstash/7.x.nix:12]: I am not sure why we need the let in here. Do I miss something?

Comment (roberth) [reply pkgs/tools/misc/logstash/7.x.nix:12]: It's for the assertion in `passthru.tests`.

**Pull request merged**